Source/Core/DolphinQt/GCMemcardCreateNew.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt/GCMemcardCreateNew.h"
+
+#include <QComboBox>
+#include <QFileDialog>
+#include <QGridLayout>
+#include <QLabel>
+#include <QPushButton>
+#include <QRadioButton>
+
+#include "Common/FileUtil.h"
+#include "Common/MsgHandler.h"
+
+#include "Core/HW/GCMemcard/GCMemcard.h"
+
+GCMemcardCreateNew::GCMemcardCreateNew(QWidget* parent) : QDialog(parent)
+{
+  m_combobox_size = new QComboBox();
+  m_combobox_size->addItem(tr("4 Mbit (59 blocks)"), QVariant(4));
+  m_combobox_size->addItem(tr("8 Mbit (123 blocks)"), QVariant(8));
+  m_combobox_size->addItem(tr("16 Mbit (251 blocks)"), QVariant(16));
+  m_combobox_size->addItem(tr("32 Mbit (507 blocks)"), QVariant(32));
+  m_combobox_size->addItem(tr("64 Mbit (1019 blocks)"), QVariant(64));
+  m_combobox_size->addItem(tr("128 Mbit (2043 blocks)"), QVariant(128));
+  m_combobox_size->setCurrentIndex(5);
+
+  m_radio_western = new QRadioButton(QString(tr("Western (Windows-1252)")));
+  m_radio_shiftjis = new QRadioButton(QString(tr("Japanese (Shift-JIS)")));
+  m_radio_western->setChecked(true);
+
+  QLabel* card_size_label = new QLabel(QString(tr("Card Size")));
+  QLabel* card_encoding_label = new QLabel(QString(tr("Encoding")));
+  QPushButton* create_card_button = new QPushButton(QString(tr("Create...")));
+
+  QGridLayout* layout = new QGridLayout();
+  layout->addWidget(card_size_label, 0, 0);
+  layout->addWidget(m_combobox_size, 0, 1);
+  layout->addWidget(card_encoding_label, 1, 0);
+  layout->addWidget(m_radio_western, 1, 1);
+  layout->addWidget(m_radio_shiftjis, 2, 1);
+  layout->addWidget(create_card_button, 3, 1, Qt::AlignRight);
+  setLayout(layout);
+
+  connect(create_card_button, &QPushButton::clicked, [this] {
+    if (CreateCard())
+      accept();
+  });
+
+  setWindowTitle(tr("Create New Memory Card"));
+  setWindowFlags(windowFlags() & ~Qt::WindowContextHelpButtonHint);
+}
+
+GCMemcardCreateNew::~GCMemcardCreateNew() = default;
+
+bool GCMemcardCreateNew::CreateCard()
+{
+  u16 size = static_cast<u16>(m_combobox_size->currentData().toInt());
+  bool is_shift_jis = m_radio_shiftjis->isChecked();
+
+  QString path = QFileDialog::getSaveFileName(
+      this, tr("Create New Memory Card"), QString::fromStdString(File::GetUserPath(D_GCUSER_IDX)),
+      tr("GameCube Memory Cards (*.raw *.gcp)") + QStringLiteral(";;") + tr("All Files (*)"));
```

Comment:
Having `*.raw` and `*.gcp` next to each other without a separator (like a comma or semicolon) seems odd. Does this affect the dialog in anyway wrt filtering existing files?

Reply:
That line is basically copied from here so if it's wrong we should change it there too: https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/DolphinQt/GCMemcardManager.cpp#L292
I just did a quick test and the dialog shows both *.raw and *.gcp files with the current filter.

And yeah, this affects the files it shows, and the default extension it attaches to the filename if you provide none, `.raw` in this case.